vault/core.go

```diff
@@ -657,23 +657,24 @@ func (c *Core) checkToken(req *logical.Request) (*logical.Auth, *TokenEntry, err
 		}
 	}
 
+	// Create the auth response
+	auth := &logical.Auth{
+		ClientToken: req.ClientToken,
+		Policies:    te.Policies,
+		Metadata:    te.Meta,
+		DisplayName: te.DisplayName,
+	}
+
 	// Check the standard non-root ACLs. Return the token entry if it's not
 	// allowed so we can decrement the use count.
 	allowed, rootPrivs := acl.AllowOperation(req)
 	if !allowed {
-		return nil, te, logical.ErrPermissionDenied
+		return auth, te, logical.ErrPermissionDenied
```

Comment:
Sure.  I've got another push coming to add populating the token and accessor fields in the auth object as well.  (see latest comments on https://github.com/hashicorp/vault/issues/2741) I'll include that in the push.